### PR TITLE
Incremental roll out

### PR DESCRIPTION
**Pull request by Nitsorn**
Client side of incremental rollout feature
- Created incrementalRolloutService
- user on login will call the service to fetch available features from streamlabs api
- user on logout will clear persistent state for available features
- hide/ show feature based on availability (TopNav for example)
```
// possible available features
export enum AvailableFeatures {
  chatbot = 'slobs--chatbot'
}
```

### CODE REVIEW

File: app/services/incrementalRollout.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+
+import { PersistentStatefulService } from 'services/persistent-stateful-service';
+import { Inject } from 'util/injector';
+import { handleErrors, authorizedHeaders } from 'util/requests';
+import { mutation } from 'services/stateful-service';
+import { UserService } from 'services/user';
+import { HostsService } from './hosts';
+
+
+export enum AvailableFeaturesEnum {
```
Comment (holiber) [inline app/services/incrementalRollout.ts]: AvailableFeaturesEnum -> EAvailableFeatures

File: app/services/user.ts [hunk 2]
```diff
@@ -223,6 +226,8 @@ export class UserService extends PersistentStatefulService<IUserServiceState> {
     await this.sceneCollectionsService.safeSync();
     // signs out of chatbot
     await this.chatbotApiService.logOut();
+    // clear available features
+    await this.incrementalRolloutService.clearAvailableFeatures();
```
Comment (holiber) [inline app/services/user.ts]: What is the goal of using PersistanceStatefulService if you always call `clearAvailableFeatures`?

**Comment by Nitsorn** [comment]
not merging in until feature freeze is complete

**Comment by Nitsorn** [comment]
chatbot tab is still commented out. Will need to coordinate re-enabling when the backend for incremental-rollout feature is deployed on sl.com https://github.com/twitchalerts/twitchalerts/pull/5268#pullrequestreview-150767420

**Pull request merged**